src/app/shared/form/form.service.ts

```diff
@@ -105,6 +108,32 @@ export class FormService {
     });
   }
 
+  public hasValidationErrors(formGroup: FormGroup | FormArray): boolean {
```

Comment:
Could we add some TypeDocs to this method & all other new methods in this class?

Reply:
done